include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -602,13 +603,13 @@ namespace CUDAWrappers
            cells_per_block_shmem(int dim, int fe_degree)
   {
     /* clang-format off */
-    return dim==2 ? (fe_degree==1 ? 32 :
-                     fe_degree==2 ? 8 :
-                     fe_degree==3 ? 4 :
-                     fe_degree==4 ? 4 :
+    return dim==2 ? (fe_degree==1 ? CUDAWrappers::warp_size :
+                     fe_degree==2 ? CUDAWrappers::warp_size/4 :
+                     fe_degree==3 ? CUDAWrappers::warp_size/8 :
+                     fe_degree==4 ? CUDAWrappers::warp_size/8 :
                      1) :
-           dim==3 ? (fe_degree==1 ? 8 :
-                     fe_degree==2 ? 2 :
+           dim==3 ? (fe_degree==1 ? CUDAWrappers::warp_size/4 :
+                     fe_degree==2 ? CUDAWrappers::warp_size/8 :
```

Comment:
Should this be `warp_size/16` or was the original incorrect?

Reply:
So we quickly looked at these numbers and they are OK but maybe not the best. @masterleinad was able to get a 10% increase of performance by tweaking the numbers but it didn't behave like we expected. So for now we stick with what we have.

Reply:
Oops sorry. You are right it should be 16 not 8.